db/common/src/main/java/io/helidon/db/common/AbstractDbExecute.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.db.common;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+
+import io.helidon.db.DbExecute;
+import io.helidon.db.DbResult;
+import io.helidon.db.DbRow;
+import io.helidon.db.DbRowResult;
+import io.helidon.db.DbStatement;
+import io.helidon.db.DbStatementType;
+import io.helidon.db.DbStatements;
+
+/**
+ * Implements methods that do not require implementation for each provider.
+ */
+public abstract class AbstractDbExecute implements DbExecute {
+    private final DbStatements statements;
+
+    /**
+     * Create an instance with configured statements.
+     *
+     * @param statements statements to obtains named statements, esp. for {@link #statementText(String)}.
+     */
+    protected AbstractDbExecute(DbStatements statements) {
+        this.statements = statements;
+    }
+
+    /**
+     * Return a statement text based on the statement name.
+     * This is a utility method that probably would use {@link io.helidon.db.DbStatements} to retrieve the named statements.
+     *
+     * @param name name of the statement
+     * @return statement text
+     */
+    protected String statementText(String name) {
+        return statements.statement(name);
+    }
+
+    @Override
```

Comment:
All those implementations depend on wrong prototype in API interface. This shall be rewritten.